ADR/0035-artifact-streams.md

```diff
@@ -0,0 +1,150 @@
+supply chain. It not only defines the storage location of the object itself but also the location of any
+additional supporting metadata like produced SBOMs, provenance, and signatures (and the public gpg keys).
+
+Since the concept of an `Artifact` exists outside of any other controller, the object may either be manually
+created based on known artifacts or a constructor may be leveraged to watch for specific actions (i.e. within
+the kubernetes cluster or new artifacts being pushed to an external location) in order to produce the resulting
+`Artifact`. An example of this constructor can resemble the `TestSubjectConstructor` as proposed in
+[Upstreaming integration-service].
+
+`Artifact`s can also store references to the process which created the object (i.e. PipelineRun). As soon as a
+PipelineRun is created, an `Artifact` should be created to enable visibility into the ongoing process if
+an `ArtifactStream` is used for tracking or visualization.
+
+### ArtifactStream
+
+All `Artifact`s have an owner reference to a single `ArtifactStream`. The `ArtifactStream` keeps track of the latest
+produced `Artifact` that has been created owned by that stream (i.e. similar to a floating tag reference).
+
+A validation webhook will prevent an `Artifact` from being created unless its configuration matches that of its owning
+`ArtifactStream`.
+
+### ArtifactCollection
+
+An `ArtifactCollection` is a set of `Artifact`s that are grouped together. An `ArtifactCollection` can include
+references to other `ArtifactCollection`s to enable intermediate logical separation and aggregated groupings.
+Additionally, there are no restrictions around how many `ArtifactCollection`s a given `Artifact` may belong to.
+
+### ArtifactCollectionStream
+
+All `ArtifactCollection`s have an owner reference to a single `ArtifactCollectionStream`. The `ArtifactCollectionStream`
+defines the set of `ArtifactStream`s and `ArtifactCollectionStream`s that are allowed in the owned `ArtifactCollection`s.
+The `ArtifactCollectionStream` also defines whether any individual update to a contained stream results in the automatic
+generation of a new owned `ArtifactCollection`. The `ArtifactCollectionStream` keeps track of the latest produced
+`ArtifactCollection` owned by that stream.
+
+A validation webhook will prevent an `ArtifactCollection` from being created unless all contained `Artifact`s and
+`ArtifactCollection`s match its owning `ArtifactCollectionStream`.
+
+## Consequences
+
+- If any controller does not want to interface with the `Artifact*` CRs directly (i.e. to achieve independence
+  from other controllers within AppStudio), they will need to create an adapter pattern between the relevant
+  APIs and their own internal data model.
+- We should now have native ability to describe built artifact and their various supply chain related attestations.
+  This ability holds whether the artifact is built within the kubernetes cluster or outside of it.
```

Comment:
Do you mean that a user can build an artifact somewhere and add a record of it in the system?